packages/portfolio-contract/src/portfolio.flows.ts

```diff
@@ -570,6 +607,78 @@ const addToUSDNPosition = async (
   ]);
 };
 
+const withdrawFromUSDNPosition = async (
+  ctx: PortfolioInstanceContext,
+  seat: ZCFSeat,
+  pos: USDNPosition,
+  kit: GuestInterface<PortfolioKit>,
+  usdcOut: bigint = 0n,
+) => {
+  const ica = kit.reader.getNobleICA() as unknown as NobleAccount;
+  const localAcct = kit.reader.getLCA() as unknown as LocalAccount;
+
+  const usdcBrand = pos.getBrand(); // Assume this gets the brand of USDC
+  const denom = 'uusdc';
+  const volume: DenomAmount = { denom, value: usdcOut };
+  const amount: Amount<'nat'> = { brand: usdcBrand, value: usdcOut };
+
+  await trackFlow(kit.reporter, [
+    {
+      how: 'Unlock, Swap',
+      src: { pos },
+      dest: { account: ica },
+      amount,
+      apply: async () => {
```

Comment:
this code is a mirror of the array in `addTo`. I **suggest** we dry it out. Make a function to take the deposit array and flip it around for withdraw, perhaps?